### PR TITLE
[screengrab] added new internal path for screenshots and run adb root to prevent access denied

### LINKED ISSUE #13787: [Screengrab] succesfully creates screenshots but looks in the wrong folder for them
### Question Checklist

- [x] Updated fastlane to the latest version
- [ ] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Question Subject

Fastlane screengrab is creating screenshots and copies them to my computer, but than tells me that there are no screenshots.

### Question Description

* When screengrab is finishing it copies the screenshots to my computer with `adb pull` from `/sdcard/my.app/screengrab/` successfully.
* Than it wants to get the screenshots again from `/data/data/my.app/app_screengrab/`. This fails because the folder does not exist.
* The logcat tells me, that screengrab is using `/storage/emulated/0/my.app/screengrab/` as directory to save the pngs to.

Why is screengrab looking into the /data/data folder? Why isn't it using one of the other two paths? Can I prevent this somehow?

### Logs

Logcat (Excerpt):
```
2018-11-30 10:54:15.788 5537-5564/my.app D/Screengrab: Using screenshot storage directory: /storage/emulated/0/my.app/screengrab/fr-CA/images/screenshots
2018-11-30 10:54:15.905 5537-5564/my.app D/Screengrab: Captured screenshot "main.png"

```
 Fastlane:

```
➜  myApp git:(hb/screenshots) ✗ fastlane screengrab --locales 'fr-CA'
[✔] 🚀
[10:15:25]: fastlane detected a Gemfile in the current directory
[10:15:25]: however it seems like you don't use `bundle exec`
[10:15:25]: to launch fastlane faster, please use
[10:15:25]:
[10:15:25]: $ bundle exec fastlane screengrab --locales fr-CA
[10:15:25]:
[10:15:25]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
[10:15:25]: Successfully loaded '/Users/henning.bunk/repos/android/wetter-app/Screengrabfile' 📄

+-----------------------------+-------------------------------------------------------------------------------------------------------------------------+
|                                                        Detected Values from './Screengrabfile'                                                        |
+-----------------------------+-------------------------------------------------------------------------------------------------------------------------+
| launch_arguments            | ["isScreengrab true"]                                                                                                   |
| build_tools_version         | 27.0.1                                                                                                                  |
| reinstall_app               | false                                                                                                                   |
| app_apk_path                | wetterApp/build/outputs/apk/myApp/uitest/my-App-uitest.apk                         |
| tests_apk_path              | wetterApp/build/outputs/apk/androidTest/myApp/uitest/my-App-uitest-androidTest.apk |
| app_package_name            | my.app                                                                                            |
| tests_package_name          | my.app.test                                                                                                    |
| use_tests_in_packages       | ["my.app.tests"]                                                                                     |
| test_instrumentation_runner | my.app.RxTestRunner                                                                                  |
| output_directory            | fastlane/screengrab                                                                                                     |
| clear_previous_screenshots  | false                                                                                                                   |
| skip_open_summary           | true                                                                                                                    |
+-----------------------------+-------------------------------------------------------------------------------------------------------------------------+


+-----------------------------+-------------------------------------------------------------------------------------------------------------------------+
|                                                            Summary for screengrab 2.108.0                                                             |
+-----------------------------+-------------------------------------------------------------------------------------------------------------------------+
| locales                     | ["fr-CA"]                                                                                                               |
| build_tools_version         | 27.0.1                                                                                                                  |
| clear_previous_screenshots  | false                                                                                                                   |
| output_directory            | fastlane/screengrab                                                                                                     |
| skip_open_summary           | true                                                                                                                    |
| app_package_name            | my.app                                                                                            |
| tests_package_name          | my.app.test                                                                                                    |
| use_tests_in_packages       | ["my.app.tests"]                                                                                     |
| launch_arguments            | ["isScreengrab true"]                                                                                                   |
| test_instrumentation_runner | my.app.RxTestRunner                                                                                  |
| ending_locale               | en-US                                                                                                                   |
| app_apk_path                | app/build/outputs/apk/myApp/uitest/my-App-uitest.apk                         |
| tests_apk_path              | app/build/outputs/apk/androidTest/myApp/uitest/my-App-uitest-androidTest.apk |
| device_type                 | phone                                                                                                                   |
| exit_on_test_failure        | true                                                                                                                    |
| reinstall_app               | false                                                                                                                   |
+-----------------------------+-------------------------------------------------------------------------------------------------------------------------+

[10:15:26]: $ /Users/henning.bunk/Library/Android/sdk/platform-tools/adb devices -l
[10:15:26]: ▸ List of devices attached
[10:15:26]: ▸ emulator-5554          device product:sdk_gphone_x86 model:Android_SDK_built_for_x86 device:generic_x86 transport_id:5
[10:15:26]: $ /Users/henning.bunk/Library/Android/sdk/platform-tools/adb -s emulator-5554 shell echo \$EXTERNAL_STORAGE
[10:15:26]: ▸ /sdcard
[10:15:26]: Cleaning screenshots on device
ls: /sdcard/my.app/screengrab: No such file or directory
[10:15:26]: Exit status: 1
ls: /data/data/my.app/app_screengrab: No such file or directory
[10:15:26]: Exit status: 1
[10:15:26]: Validating app APK
[10:15:26]: $ /Users/henning.bunk/Library/Android/sdk/build-tools/28.0.3/aapt dump permissions wetterApp/build/outputs/apk/myApp/uitest/my-App-uitest.apk
[10:15:26]: ▸ package: my.app
[10:15:26]: ▸ uses-permission: name='android.permission.DISABLE_KEYGUARD'
[10:15:26]: ▸ uses-permission: name='android.permission.WAKE_LOCK'
[10:15:26]: ▸ uses-permission: name='android.permission.WRITE_EXTERNAL_STORAGE'
[10:15:26]: ▸ uses-permission: name='android.permission.READ_EXTERNAL_STORAGE'
[10:15:26]: ▸ uses-permission: name='android.permission.CHANGE_CONFIGURATION'
[10:15:26]: ▸ uses-permission: name='android.permission.ACCESS_FINE_LOCATION'
[10:15:26]: ▸ uses-permission: name='android.permission.ACCESS_COARSE_LOCATION'
[10:15:26]: ▸ uses-permission: name='android.permission.INTERNET'
[10:15:26]: ▸ uses-permission: name='android.permission.ACCESS_NETWORK_STATE'
[10:15:26]: ▸ uses-permission: name='com.google.android.providers.gsf.permission.READ_GSERVICES'
[10:15:26]: ▸ uses-permission: name='android.permission.VIBRATE'
[10:15:26]: ▸ uses-permission: name='android.permission.CAMERA'
[10:15:26]: ▸ uses-permission: name='android.permission.RECEIVE_BOOT_COMPLETED'
[10:15:26]: ▸ uses-permission: name='com.android.vending.BILLING'
[10:15:26]: ▸ uses-permission: name='android.permission.ACCESS_WIFI_STATE'
[10:15:26]: ▸ uses-permission: name='com.google.android.finsky.permission.BIND_GET_INSTALL_REFERRER_SERVICE'
[10:15:26]: ▸ uses-permission: name='com.google.android.c2dm.permission.RECEIVE'
[10:15:26]: Installing app APK
[10:15:26]: $ /Users/henning.bunk/Library/Android/sdk/platform-tools/adb -s emulator-5554 install -t -r wetterApp/build/outputs/apk/myApp/uitest/my-App-uitest.apk
[10:15:28]: ▸ Success
[10:15:28]: Installing tests APK
[10:15:28]: $ /Users/henning.bunk/Library/Android/sdk/platform-tools/adb -s emulator-5554 install -t -r wetterApp/build/outputs/apk/androidTest/myApp/uitest/my-App-uitest-androidTest.apk
[10:15:30]: ▸ Success
[10:15:30]: Granting the permission necessary to change locales on the device
[10:15:30]: $ /Users/henning.bunk/Library/Android/sdk/platform-tools/adb -s emulator-5554 shell pm grant my.app android.permission.CHANGE_CONFIGURATION
[10:15:30]: $ /Users/henning.bunk/Library/Android/sdk/platform-tools/adb -s emulator-5554 shell getprop ro.build.version.sdk
[10:15:30]: ▸ 28
[10:15:30]: Granting the permissions necessary to access device external storage
[10:15:30]: $ /Users/henning.bunk/Library/Android/sdk/platform-tools/adb -s emulator-5554 shell pm grant my.app android.permission.WRITE_EXTERNAL_STORAGE
[10:15:30]: $ /Users/henning.bunk/Library/Android/sdk/platform-tools/adb -s emulator-5554 shell pm grant my.app android.permission.READ_EXTERNAL_STORAGE
[10:15:30]: Running tests for locale: fr-CA
[10:15:30]: $ /Users/henning.bunk/Library/Android/sdk/platform-tools/adb -s emulator-5554 shell am instrument --no-window-animation -w \
-e testLocale fr_CA \
-e endingLocale en_US \
-e package my.app.tests \
-e isScreengrab true \
my.app.test/my.app.RxTestRunner
[10:15:41]: ▸ my.app.tests.MainScreenshots:
[10:16:27]: ▸ my.app.tests.MainScreenshots:....
[10:16:30]: ▸ Time: 55.69
[10:16:30]: ▸ OK (9 tests)
[10:16:30]: Pulling captured screenshots from the device
[10:16:30]: $ /Users/henning.bunk/Library/Android/sdk/platform-tools/adb -s emulator-5554 pull /sdcard/my.app/screengrab /var/folders/1z/r9tnh30d2295kv_vs7q1f47r0000gp/T/d20181130-3871-1ty3xa1
ls: /data/data/my.app/app_screengrab: No such file or directory
[10:16:31]: Exit status: 1
[10:16:31]: Screenshots copied to fastlane/fr-CA/images/phoneScreenshots
[10:16:31]: Make sure you've used Screengrab.screenshot() in your tests and that your expected tests are being run.

[!] No screenshots were detected 📷❌
```

### PR DESCRIPTION
**Pull request by joshdholtz**
## Motivation and Context
Fixes #13787
Screenshots were at a different path and getting access denied

### Description
- Added another internal path to look at for screenshots
  - Got this path by looking at logical
- Started getting permission denied on this path
  - Turned out need to restart `adb` by running `adb root` if trying to access internal storage like `/data`

### CODE REVIEW

File: screengrab/lib/screengrab/runner.rb [hunk 1]
```diff
@@ -146,8 +149,11 @@ def determine_external_screenshots_path(device_serial)
       File.join(device_ext_storage, @config[:app_package_name], 'screengrab')
     end
 
-    def determine_internal_screenshots_path
-      "/data/data/#{@config[:app_package_name]}/app_screengrab"
+    def determine_internal_screenshots_paths
+      [
+        "/data/data/#{@config[:app_package_name]}/app_screengrab",
```
Comment (snatchev) [inline screengrab/lib/screengrab/runner.rb]: Please put `app_package_name` as a function parameter, passed in by the caller.

File: screengrab/lib/screengrab/runner.rb [hunk 2]
```diff
@@ -146,8 +149,11 @@ def determine_external_screenshots_path(device_serial)
...
+        "/data/user/0/#{@config[:app_package_name]}/files/#{@config[:app_package_name]}/screengrab//en-US/images/screenshots"
```
Comment (snatchev) [inline screengrab/lib/screengrab/runner.rb]: probably not an issue, but there's two slashes `//en-US`

**Review by snatchev — CHANGES_REQUESTED** [review]

File: screengrab/lib/screengrab/runner.rb (hunk 1 above)
Comment (joshdholtz) [reply screengrab/lib/screengrab/runner.rb]: Smart 👌

File: screengrab/lib/screengrab/runner.rb (hunk 2 above)
Comment (joshdholtz) [reply screengrab/lib/screengrab/runner.rb]: Whooooopppss

**Review by snatchev — APPROVED** [review]
LGTM

**Pull request merged**